### PR TITLE
RFC: E2E test for GlusterFS

### PR DESCRIPTION
**Pull request by jsafrane**
Based on my NFS test in PR #6755.  It has the same problems as the  NFS test - it needs "random" container which exports a file over GlusterFS. Based on Fedora, ~400MB.

I'd really like to see some policy about these test containers...

### CODE REVIEW

File: cluster/saltbase/salt/base.sls [hunk 1]
```diff
@@ -5,7 +5,9 @@ pkg-core:
 {% if grains['os_family'] == 'RedHat' %}
       - python
       - git
+      - glusterfs-fuse
```
Comment (timothysc) [inline cluster/saltbase/salt/base.sls]: Do we want to do this in the base, or explicitly option enable?

File: test/e2e/nfs.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+	// filled in BeforeEach
+	var c *client.Client
+	var namespace string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+		namespace = "e2e-ns-" + dateStamp() + "-0"
```
Comment (timothysc) [inline test/e2e/nfs.go]: I believe there are new mechanics for this, see @jayunit100

File: test/e2e/nfs.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+				Name: "nfs-server-" + string(util.NewUUID()),
+				Labels: map[string]string{
+					"role": "nfs-server",
+				},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:       "nfs-server",
+						Image:      "jsafrane/nfs-data",
```
Comment (timothysc) [inline test/e2e/nfs.go]: @satnam6502 @zmerlynn - how can we get e2e images up in standard repo?

File: test/e2e/nfs.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+		namespace = "e2e-ns-" + dateStamp() + "-0"
+	})
+
+	AfterEach(func() {
+		if clean {
+			c.Namespaces().Delete(namespace)
+		}
+	})
+
+	It("should be able to mount NFS", func() {
```
Comment (timothysc) [inline test/e2e/nfs.go]: We should probably skip by default and we can option enable.

File: test/e2e/nfs.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NFS", func() {
```
Comment (timothysc) [inline test/e2e/nfs.go]: Personal Nit - breaking up into functions would be nice.

File: test/e2e/nfs.go (hunk 2 above)
Comment (jayunit100) [inline test/e2e/nfs.go]: Indeed now you can use the create e2e namespace uuid method in util.go.

As an example see the nenetworking.go tests

Comment (jsafrane) [inline test/e2e/nfs.go]: Ok, I'll do.

File: cluster/saltbase/salt/base.sls (hunk 1 above)
Comment (jsafrane) [inline cluster/saltbase/salt/base.sls]: IMO there is no option to exclude glusterfs from build, so any Kubernetes release will have support for it. Gluster should be part of the default installation in the same way as e.g. nfs-utils inherited from the base image.

File: test/e2e/nfs.go (hunk 5 above)
Comment (jsafrane) [inline test/e2e/nfs.go]: Ok.

**Comment by jsafrane** [comment]
Let's continue in PR #7435. I think I addressed all the comments there.

**Pull request closed without merging**